File: flutter/diary_app/lib/repository/diary_repository.dart
import 'package:diary_app/domain/diary.dart';

class DiaryRepository {

  List<Diary> pagesOfDiary ;
  DiaryRepository(this.pagesOfDiary);


  Diary findOne(String id){
    Diary diary = Diary("", "2021-01-01", "", "", "");

    for (int i = 0; i < pagesOfDiary.length; i++){
      if(pagesOfDiary[i].day == id) {
        diary = pagesOfDiary[i];
      }
    }
    return diary;
  }

  List<Diary> findAll() {
    return pagesOfDiary;
  }

  void save(Diary diary) {
    pagesOfDiary.add(diary);
  }

  void update(Diary diary) {
    Diary diaryUpdate = findOne(diary.day);

    diaryUpdate.color = diary.color;
    diaryUpdate.summary = diary.summary;
    diaryUpdate.text = diary.text;
  }

  void delete(Diary diary) {
    Diary diaryDelete = findOne(diary.day);

    pagesOfDiary.remove(diaryDelete);
  }

}

File: android/DiaryApp/app/src/main/java/ma/android/diaryapp/repository/DiaryRepository.kt
package ma.android.diaryapp.repository

import androidx.annotation.WorkerThread
import androidx.lifecycle.LiveData
import ma.android.diaryapp.data.domain.Diary
import ma.android.diaryapp.database.dao.DiaryDao


class DiaryRepository(private val diaryDao: DiaryDao) {

    var pagesOfDiary: LiveData<List<Diary>> = diaryDao.findAll()

    @Suppress("RedundantSuspendModifier")
    @WorkerThread
    suspend fun save(diary: Diary) {
        diaryDao.insert(diary)
    }

    @Suppress("RedundantSuspendModifier")
    @WorkerThread
    suspend fun update(diary: Diary) {
        diaryDao.update(diary)
    }

    @Suppress("RedundantSuspendModifier")
    @WorkerThread
    suspend fun delete(diary: Diary) {
        diaryDao.delete(diary)
    }

}

